lib/routes/account.js

```diff
@@ -549,6 +549,61 @@ module.exports = function (
     },
     {
       method: 'GET',
+      path: '/account/profile',
+      config: {
+        auth: {
+          mode: 'optional',
+          strategies: [
+            'sessionToken',
+            'oauthToken'
+          ]
+        }
+      },
+      handler: function (request, reply) {
+        var auth = request.auth
+        var uid
+        if (auth.strategy === 'sessionToken') {
+          uid = auth.credentials.uid
+        } else {
+          uid = Buffer(auth.credentials.user, 'hex')
+        }
+        function hasProfileItemScope(item) {
+          if (auth.strategy === 'sessionToken') {
+            return true
+          }
+          var scopes = auth.credentials.scopes
+          for (var i = 0; i < scopes.length; i++) {
+            if (scopes[i] === 'profile') {
```

Comment:
How about:

```js
var allowed = set('profile', 'profile:write', 'profile:' + item, 'profile:' + item + ':write');
for (var i = 0, len = scopes.length; i < len; i++) {
  if (allowed[scopes[i]]) {
    return true;
  }
}
```

Assuming `set` is some sort of function to easily make an object:

```js
function set() {
  var obj = {};
  for (var i = 0, len = arguments.length; i < len; i++) {
    obj[arguments[i]] = 1;
  }
  return obj;
}
```

Bonus points if these sets are `const`s created once, instead of every request.